R/science_article.R

```diff
@@ -0,0 +1,482 @@
+#' Science Journal Format
+#'
+#' Format for creating submissions to Science. Based on the Science
+#' \href{https://www.sciencemag.org/site/feature/contribinfo/prep/TeX_help/#downloads}{class}.
+#'
+#' @inheritParams rmarkdown::pdf_document
+#' @param ... Additional arguments to \code{rmarkdown::pdf_document}
+#' @param move_figures set to \code{TRUE} to move figures to end. Default is \code{TRUE}. Only
+#' works when `draft == TRUE`.
+#' @param move_tables set to \code{TRUE} to move tables to end. Default is \code{TRUE}. Only
+#' works when `draft == TRUE`.
+#' @param draft set to \code{TRUE} for the draft version or \code{FALSE} for a
+#' final submission version. \code{TRUE} creates a sub directory in the document
+#' directory and saves each figure to its own PDF there. It also moves the supplemental
+#' materials to its own file.
+#' @md
+#' @export
+science_article <- function(..., keep_tex = TRUE, move_figures = TRUE,
+                            move_tables = TRUE, number_sections = FALSE,
+                            draft = TRUE) {
+  base <- pdf_document_format(
+    "science", keep_tex = keep_tex, number_sections = number_sections, ...
+  )
+
+  # Build from the rjournal_article post processing
+  base$pandoc$to <- "latex"
+  base$pandoc$ext <- ".tex"
+
+  # Process authors;
+  # if version == 'draft' move figures/tables to the end; Remove Section Numbers
+  # if version == 'final', save each figure/table in a separate PDF &
+  # Also the Supplementary materials must be in a separate PDF
+  base$post_processor <- function(metadata, utf8_input, output_file, clean, verbose) {
+    filename <- basename(output_file)
+    # underscores in the filename will be problematic in \input{filename};
+    # pandoc will escape underscores but it should not, i.e., should be
+    # \input{foo_bar} instead of \input{foo\_bar}
+    if (filename != (filename2 <- gsub('_', '-', filename))) {
+      file.rename(filename, filename2); filename <- filename2
+    }
+
+    # post process TEX file
+    temp_tex <- xfun::read_utf8(filename)
+    temp_tex <- post_process_authors_and(temp_tex)
+    if (isTRUE(draft)) {
+      if (move_figures) {
+        temp_tex <- relocate_figures(temp_tex)
+      }
+      if (move_tables) {
+        temp_tex <- relocate_tables(temp_tex)
+      }
+    } else {
+      temp_tex <- separate_appendix(output_file, temp_tex, number_sections)
+
+      # Build Supplement
+      if (file.exists(paste0('supplement_', output_file))){
+        tinytex::latexmk(paste0('supplement_', filename),
+                         base$pandoc$latex_engine, clean = clean)
+      }
+    }
+
+    if (!number_sections) {
+      temp_tex <- unnumber_sections(temp_tex)
+    }
```

Comment:
I understand the purpose here is to have all the section unnumbered right ? 

Just a reminder this can be done in Markdown with 
```markdown
# section {.unnumbered}
```
and this could be made a default by using a Lua filter to add the class on each header that needs it. 

But I guess tex post processing is working well too

Reply:
Yeah, it's much simpler to just force this in Lua. Just updated to do that instead.